### ISSUE TITLE
default namespace didnot work in rancher 2.2 catalog apps page - UI

### ISSUE THREAD

**Issue by guangbochen** [rancher/rancher#19697]
**What kind of request is this (question/bug/enhancement/feature request):**
UI Bug

**Steps to reproduce (least amount of steps as possible):**
1. Go to the Catalog Apps page
2. Click launch and select the longhorn app
3.  The default namespace should be `longhorn-system` as it has pre-defined namespace value. `"requiredNamespace: "longhorn-system""`. 

**Result:**
set default namespace to pre-defined namespace.

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): master
- Installation option (single install/HA): HA/Single

**Comment by gabrielcalderon** [comment]
Are you getting a random string prefix to the namespace? I noticed this hapoens to me if I launch the app as a MultiCluster. 

Gets worst since the workload also gets a random string prefixed.

**Comment by alena1108** [comment]
@gabrielcalderon multi cluster app is different; today it doesn't support custom namespace on API/backend level. The issue is being tracked here: https://github.com/rancher/rancher/issues/17913

Regular app does have support for a custom namespace on the backend, and we have to fix something in the UI to make it work.

**Comment by gabrielcalderon** [comment]
@alena1108 That explains the behavior on my cluster. Thanks!

**Comment by westlywright** [comment]
@davidnuzik I've merged this into master, I'd like to get it QA'd in master before I backport the fix for the 2.2.3 target.

**Comment by davidnuzik** [comment]
Version: Master (4/22/19)

Followed the steps to reproduce. Noticed in the UI we cannot modify the requiredNamespace. Noticed upon submit we send the proper Namespace `targetNamespace: "longhorn-system"`
Noticed the app successfully provisioned without any issues. I can access index.html without any issues.

@westlywright done validating in master. It looks good.

**Comment by yasker** [comment]
v2.2.4-rc6, the app name is still randomly generated. Namespace issue has been fixed.

<img width="1297" alt="Screen Shot 2019-05-22 at 12 42 10 AM" src="https://user-images.githubusercontent.com/3344618/58135452-e725de00-7c2a-11e9-842c-2a198b7ec6c5.png">

**Comment by westlywright** [comment]
@yasker the name is unique and is randomized because of this. if you remove the random string you'll see the api will fail. not much the UI can do about this.

**Comment by yasker** [comment]
@westlywright Hmm, in fact if I removed the random string part, it works in my demo… I don't have other apps with the same name in the environment.

**Comment by westlywright** [comment]
@sheng-liang So no other apps with the longhorn-system name? just another namespace?

**Comment by yasker** [comment]
No, I only installed Longhorn once

**Comment by davidnuzik** [comment]
**Replication:**
v2.2.3

I replicated the issue where the Namespace field is editable (it should not be as it is a required namespace). I did have a hard time reproducing the issue with the random chars at the end of the name/namespace though.

![image](https://user-images.githubusercontent.com/45179589/58597104-6323b580-822b-11e9-90d0-f3fa54d8a2ce.png)

When I already had an existing namespace I did see the random chars at the end (there was a bug in v2.2.3 and earlier v2.2.4 where namespace is in "Activating" state for an abnormally long time it's possible this was issue earlier; I am not certain.

![image](https://user-images.githubusercontent.com/45179589/58597162-9b2af880-822b-11e9-8360-ae0901e1bd4a.png)


**Validation:**
Master (v2.3) (5/29/19)

In latest master the Required Namespace is now an un-editable field. I was successfully able to deploy the app.

![image](https://user-images.githubusercontent.com/45179589/58597230-db8a7680-822b-11e9-9bf3-8e971f01bea0.png)

Notice the name and namespace are "longhorn-system".

If you attempt to launch the app again to make another instance (which you are not supposed to do). The namespace remains the same, and the current "longhorn-system" namespace is utilized. I do not see an HTTP call made to attempt to create the namespace again as it already exists. The name has random chars tat the end and this is expected.

![image](https://user-images.githubusercontent.com/45179589/58597294-1db3b800-822c-11e9-9326-59992cca7839.png)

The app won't properly deploy, but this should be expected -- we should not be running two instances of the app on the same cluster (at least for longhorn and a handful of other apps).
